openedx/core/djangoapps/content/search/handlers.py

```diff
@@ -151,6 +155,24 @@ def content_library_updated_handler(**kwargs) -> None:
     update_content_library_index_docs.apply(args=[str(content_library_data.library_key)])
 
 
+@receiver(LIBRARY_COLLECTION_CREATED)
+@receiver(LIBRARY_COLLECTION_UPDATED)
+@only_if_meilisearch_enabled
+def library_collection_updated_handler(**kwargs) -> None:
+    """
+    Create or update the index for the content library collection
+    """
+    library_collection = kwargs.get("library_collection", None)
+    if not library_collection or not isinstance(library_collection, LibraryCollectionData):  # pragma: no cover
+        log.error("Received null or incorrect data for event")
+        return
+
+    update_library_collection_index_doc.delay(
+        str(library_collection.library_key),
+        library_collection.collection_key,
+    )
```

Comment:
We need to make these events sync so the frontend is updated as soon as we make a change in the search index.

```suggestion
    api.upsert_library_collection_index_doc(library_collection.library_key, library_collection.collection_key)
```

Reply:
Done with 2c6c8cf9d9.